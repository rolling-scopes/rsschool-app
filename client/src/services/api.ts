import { Configuration } from 'api';

const basePath = process.env.NODE_ENV === 'production' ? '' : 'http://localhost:3000';

export function getApiCfg() {
  return new Configuration({ basePath });
}
