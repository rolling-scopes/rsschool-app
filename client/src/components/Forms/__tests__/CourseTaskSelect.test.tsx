import React from 'react';
import { render } from '@testing-library/react';

import { CourseTaskSelect } from '..';
import { CourseTaskDto } from 'api';

const futureDate = new Date();
futureDate.setDate(futureDate.getDate() + 1);

const fullData = [
  {
    id: 432,
    taskId: 644,
    name: 'Flex / Grid test',
    maxScore: 100,
    scoreWeight: 1,
    githubPrRequired: false,
    descriptionUrl: 'https://github.com/rolling-scopes-school/tasks/blob/master/tasks/self-test.md',
    studentStartDate: '2022-02-06T13:44:00.000Z',
    studentEndDate: futureDate.toISOString(),
    useJury: false,
    checker: 'crossCheck',
    taskOwnerId: 606,
    githubRepoName: 'clean-code-s1e1',
    sourceGithubRepoUrl: 'src',
    type: 'jstask',
    special: '',
  },
  {
    id: 431,
    taskId: 729,
    name: 'Drum Kit',
    maxScore: 100,
    scoreWeight: 1,
    githubPrRequired: false,
    descriptionUrl: 'https://github.com/rolling-scopes-school/tasks/blob/master/tasks/js30/js30-1.md',
    studentStartDate: '2022-02-06T11:37:00.000Z',
    studentEndDate: futureDate.toISOString(),
    useJury: false,
    checker: 'crossCheck',
    taskOwnerId: 606,
    githubRepoName: 'clean',
    sourceGithubRepoUrl: 'src',
    type: 'jstask',
    special: '',
  },
  {
    id: 978,
    taskId: 695,
    name: 'Chess S1E2. Cross-check',
    maxScore: 480,
    scoreWeight: 1,
    githubPrRequired: false,
    descriptionUrl: 'https://github.com/rolling-scopes-school/tasks/blob/master/tasks/chess/codejam-chess-part-two.md',
    studentStartDate: '2021-06-30T00:00:00.000Z',
    studentEndDate: '2021-07-19T23:59:00.000Z',
    useJury: false,
    checker: 'crossCheck',
    taskOwnerId: 2084,
    githubRepoName: 'js',
    sourceGithubRepoUrl: 'src',
    type: 'jstask',
    special: '',
  },
  {
    id: 821,
    name: 'CV. Cross-Check',
    maxScore: 100,
    scoreWeight: 0.2,
    descriptionUrl: 'https://github.com/rolling-scopes-school/tasks/blob/master/tasks/cv/html-css.md',
    studentStartDate: '2021-02-28T10:00:00.000Z',
    studentEndDate: '2021-03-08T23:59:00.000Z',
    checker: 'crossCheck',
    taskOwnerId: 2084,
    type: 'htmltask',
  },
] as CourseTaskDto[];

const onlyExpired = [
  {
    id: 978,
    name: 'Chess S1E2. Cross-check',
    maxScore: 480,
    scoreWeight: 1,
    githubPrRequired: false,
    descriptionUrl: 'https://github.com/rolling-scopes-school/tasks/blob/master/tasks/chess/codejam-chess-part-two.md',
    studentStartDate: '2021-06-30T00:00:00.000Z',
    studentEndDate: '2021-07-19T23:59:00.000Z',
    useJury: false,
    checker: 'crossCheck',
    taskOwnerId: 2084,
    githubRepoName: 'js',
    sourceGithubRepoUrl: 'src',
    type: 'jstask',
    special: '',
  },
  {
    id: 821,
    name: 'CV. Cross-Check',
    maxScore: 100,
    scoreWeight: 0.2,
    descriptionUrl: 'https://github.com/rolling-scopes-school/tasks/blob/master/tasks/cv/html-css.md',
    studentStartDate: '2021-02-28T10:00:00.000Z',
    studentEndDate: '2021-03-08T23:59:00.000Z',
    checker: 'crossCheck',
    taskOwnerId: 2084,
    type: 'htmltask',
  },
] as CourseTaskDto[];

describe('CourseTaskSelect', () => {
  const { container: fullDataContainer } = render(
    <CourseTaskSelect data={fullData} groupBy="deadline"></CourseTaskSelect>,
  );
  const { container: expiredContainer } = render(
    <CourseTaskSelect data={onlyExpired} groupBy="deadline"></CourseTaskSelect>,
  );
  const { container: emptyContainer } = render(<CourseTaskSelect data={[]} groupBy="deadline"></CourseTaskSelect>);

  it('should render with both groups', () => {
    expect(fullDataContainer).toMatchSnapshot();
  });
  it('should render with one group', () => {
    expect(expiredContainer).toMatchSnapshot();
  });

  it('should render empty', () => {
    expect(emptyContainer).toMatchSnapshot();
  });
});
