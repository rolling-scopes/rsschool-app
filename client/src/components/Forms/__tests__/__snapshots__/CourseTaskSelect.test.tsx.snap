// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseTaskSelect Should render correctly if groupBy is cross-check deadline 1`] = `
<body>
  <div>
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-w8mnev"
    >
      <div
        class="ant-form-item css-dev-only-do-not-override-w8mnev"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-w8mnev"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-w8mnev"
          >
            <label
              class="ant-form-item-required"
              for="courseTaskId"
              title="Task"
            >
              Task
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-w8mnev"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  aria-required="true"
                  class="ant-select ant-select-in-form-item css-dev-only-do-not-override-w8mnev ant-select-single ant-select-show-arrow ant-select-open"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="courseTaskId_list_1"
                        aria-autocomplete="list"
                        aria-controls="courseTaskId_list"
                        aria-expanded="true"
                        aria-haspopup="listbox"
                        aria-owns="courseTaskId_list"
                        aria-required="true"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="courseTaskId"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Select task
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
  <div>
    <div
      class="ant-select-dropdown ant-slide-up-appear ant-slide-up-appear-prepare ant-slide-up css-dev-only-do-not-override-w8mnev ant-select-dropdown-placement-bottomLeft"
      style="--arrow-x: 0px; --arrow-y: 0px; left: -1000vw; top: -1000vh; box-sizing: border-box; min-width: 0; width: 0px;"
    >
      <div>
        <div
          id="courseTaskId_list"
          role="listbox"
          style="height: 0px; width: 0px; overflow: hidden;"
        >
          <div
            aria-selected="false"
            id="courseTaskId_list_0"
            role="presentation"
          />
          <div
            aria-selected="false"
            id="courseTaskId_list_1"
            role="option"
          >
            434
          </div>
          <div
            aria-selected="false"
            id="courseTaskId_list_2"
            role="option"
          >
            450
          </div>
        </div>
        <div
          class="rc-virtual-list"
          style="position: relative;"
        >
          <div
            class="rc-virtual-list-holder"
            style="max-height: 274px; overflow-y: hidden;"
          >
            <div
              style="height: 288px; position: relative; overflow: hidden;"
            >
              <div
                class="rc-virtual-list-holder-inner"
                style="display: flex; flex-direction: column; transform: translateY(0px); position: absolute; left: 0px; right: 0px; top: 0px;"
              >
                <div
                  class="ant-select-item ant-select-item-group"
                  title="Review (2)"
                >
                  Review (2)
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped ant-select-item-option-active"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                    <span
                      aria-label="clock-circle"
                      class="anticon anticon-clock-circle"
                      role="img"
                      style="font-size: 14px; color: rgb(255, 77, 79);"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="clock-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                        <path
                          d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                        />
                      </svg>
                    </span>
                     
                    Async Race
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                    <span
                      aria-label="clock-circle"
                      class="anticon anticon-clock-circle"
                      role="img"
                      style="font-size: 14px; color: rgb(250, 173, 20);"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="clock-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                        <path
                          d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                        />
                      </svg>
                    </span>
                     
                    Fancy-weather Cross-Check
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  class="ant-select-item ant-select-item-group"
                  title="Unknown (2)"
                >
                  Unknown (2)
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Virtual-piano
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Clean-code-s1e1
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  class="ant-select-item ant-select-item-group"
                  title="Future (2)"
                >
                  Future (2)
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Shelter Cross-check
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Virtual Keyboard Cross-Check
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  class="ant-select-item ant-select-item-group"
                  title="Expired (2)"
                >
                  Expired (2)
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Wildlife
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    CV. Cross-Check
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="rc-virtual-list-scrollbar rc-virtual-list-scrollbar-show"
            style="width: 8px; top: 0px; bottom: 0px; right: 0px; position: absolute;"
          >
            <div
              class="rc-virtual-list-scrollbar-thumb"
              style="width: 100%; height: 137px; top: 0px; left: 0px; position: absolute; background: rgba(0, 0, 0, 0.5); border-radius: 99px; cursor: pointer; user-select: none;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`CourseTaskSelect Should render correctly if groupBy is deadline 1`] = `
<body>
  <div>
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-w8mnev"
    >
      <div
        class="ant-form-item css-dev-only-do-not-override-w8mnev"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-w8mnev"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-w8mnev"
          >
            <label
              class="ant-form-item-required"
              for="courseTaskId"
              title="Task"
            >
              Task
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-w8mnev"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  aria-required="true"
                  class="ant-select ant-select-in-form-item css-dev-only-do-not-override-w8mnev ant-select-single ant-select-show-arrow ant-select-open"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="courseTaskId_list_1"
                        aria-autocomplete="list"
                        aria-controls="courseTaskId_list"
                        aria-expanded="true"
                        aria-haspopup="listbox"
                        aria-owns="courseTaskId_list"
                        aria-required="true"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="courseTaskId"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Select task
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      ,
    </form>
  </div>
  <div>
    <div
      class="ant-select-dropdown ant-slide-up-appear ant-slide-up-appear-prepare ant-slide-up css-dev-only-do-not-override-w8mnev ant-select-dropdown-placement-bottomLeft"
      style="--arrow-x: 0px; --arrow-y: 0px; left: -1000vw; top: -1000vh; box-sizing: border-box; min-width: 0; width: 0px;"
    >
      <div>
        <div
          id="courseTaskId_list"
          role="listbox"
          style="height: 0px; width: 0px; overflow: hidden;"
        >
          <div
            aria-selected="false"
            id="courseTaskId_list_0"
            role="presentation"
          />
          <div
            aria-selected="false"
            id="courseTaskId_list_1"
            role="option"
          >
            451
          </div>
          <div
            aria-selected="false"
            id="courseTaskId_list_2"
            role="option"
          >
            764
          </div>
        </div>
        <div
          class="rc-virtual-list"
          style="position: relative;"
        >
          <div
            class="rc-virtual-list-holder"
            style="max-height: 274px; overflow-y: hidden;"
          >
            <div>
              <div
                class="rc-virtual-list-holder-inner"
                style="display: flex; flex-direction: column;"
              >
                <div
                  class="ant-select-item ant-select-item-group"
                  title="Active (4)"
                >
                  Active (4)
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped ant-select-item-option-active"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                    <span
                      aria-label="clock-circle"
                      class="anticon anticon-clock-circle"
                      role="img"
                      style="font-size: 14px; color: rgb(255, 77, 79);"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="clock-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                        <path
                          d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                        />
                      </svg>
                    </span>
                     
                    Codewars Algorithms-2
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                    <span
                      aria-label="clock-circle"
                      class="anticon anticon-clock-circle"
                      role="img"
                      style="font-size: 14px; color: rgb(255, 77, 79);"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="clock-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                        <path
                          d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                        />
                      </svg>
                    </span>
                     
                    Codewars #0
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                    <span
                      aria-label="clock-circle"
                      class="anticon anticon-clock-circle"
                      role="img"
                      style="font-size: 14px; color: rgb(255, 77, 79);"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="clock-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                        <path
                          d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                        />
                      </svg>
                    </span>
                     
                    React. Testing
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                    <span
                      aria-label="clock-circle"
                      class="anticon anticon-clock-circle"
                      role="img"
                      style="font-size: 14px; color: rgb(255, 77, 79);"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="clock-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                        <path
                          d="M686.7 638.6L544.1 535.5V288c0-4.4-3.6-8-8-8H488c-4.4 0-8 3.6-8 8v275.4c0 2.6 1.2 5 3.3 6.5l165.4 120.6c3.6 2.6 8.6 1.8 11.2-1.7l28.6-39c2.6-3.7 1.8-8.7-1.8-11.2z"
                        />
                      </svg>
                    </span>
                     
                    Test Algorithms & Data structures
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="rc-virtual-list-scrollbar"
            style="width: 8px; top: 0px; bottom: 0px; right: 0px; position: absolute; display: none;"
          >
            <div
              class="rc-virtual-list-scrollbar-thumb"
              style="width: 100%; height: 137px; top: 0px; left: 0px; position: absolute; background: rgba(0, 0, 0, 0.5); border-radius: 99px; cursor: pointer; user-select: none;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`CourseTaskSelect Should render correctly if groupBy is default 1`] = `
<body>
  <div>
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-w8mnev"
    >
      <div
        class="ant-form-item css-dev-only-do-not-override-w8mnev"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-w8mnev"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-w8mnev"
          >
            <label
              class="ant-form-item-required"
              for="courseTaskId"
              title="Task"
            >
              Task
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-w8mnev"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <div
                  aria-required="true"
                  class="ant-select ant-select-in-form-item css-dev-only-do-not-override-w8mnev ant-select-single ant-select-show-arrow ant-select-open"
                >
                  <div
                    class="ant-select-selector"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-activedescendant="courseTaskId_list_1"
                        aria-autocomplete="list"
                        aria-controls="courseTaskId_list"
                        aria-expanded="true"
                        aria-haspopup="listbox"
                        aria-owns="courseTaskId_list"
                        aria-required="true"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="courseTaskId"
                        readonly=""
                        role="combobox"
                        style="opacity: 0;"
                        type="search"
                        unselectable="on"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-placeholder"
                    >
                      Select task
                    </span>
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-arrow"
                    style="user-select: none;"
                    unselectable="on"
                  >
                    <span
                      aria-label="down"
                      class="anticon anticon-down ant-select-suffix"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
  <div>
    <div
      class="ant-select-dropdown ant-slide-up-appear ant-slide-up-appear-prepare ant-slide-up css-dev-only-do-not-override-w8mnev ant-select-dropdown-placement-bottomLeft"
      style="--arrow-x: 0px; --arrow-y: 0px; left: -1000vw; top: -1000vh; box-sizing: border-box; min-width: 0; width: 0px;"
    >
      <div>
        <div
          id="courseTaskId_list"
          role="listbox"
          style="height: 0px; width: 0px; overflow: hidden;"
        >
          <div
            aria-selected="false"
            id="courseTaskId_list_0"
            role="presentation"
          />
          <div
            aria-selected="false"
            id="courseTaskId_list_1"
            role="option"
          >
            438
          </div>
          <div
            aria-selected="false"
            id="courseTaskId_list_2"
            role="option"
          >
            441
          </div>
        </div>
        <div
          class="rc-virtual-list"
          style="position: relative;"
        >
          <div
            class="rc-virtual-list-holder"
            style="max-height: 274px; overflow-y: hidden;"
          >
            <div>
              <div
                class="rc-virtual-list-holder-inner"
                style="display: flex; flex-direction: column;"
              >
                <div
                  class="ant-select-item ant-select-item-group"
                  title="All tasks (8)"
                >
                  All tasks (8)
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped ant-select-item-option-active"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Shelter Cross-check
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Virtual Keyboard Cross-Check
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Fancy-weather Cross-Check
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Async Race
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Clean-code-s1e1
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Virtual-piano
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    Wildlife
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
                <div
                  aria-selected="false"
                  class="ant-select-item ant-select-item-option ant-select-item-option-grouped"
                >
                  <div
                    class="ant-select-item-option-content"
                  >
                     
                    CV. Cross-Check
                  </div>
                  <span
                    aria-hidden="true"
                    class="ant-select-item-option-state"
                    style="user-select: none;"
                    unselectable="on"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="rc-virtual-list-scrollbar"
            style="width: 8px; top: 0px; bottom: 0px; right: 0px; position: absolute; display: none;"
          >
            <div
              class="rc-virtual-list-scrollbar-thumb"
              style="width: 100%; height: 137px; top: 0px; left: 0px; position: absolute; background: rgba(0, 0, 0, 0.5); border-radius: 99px; cursor: pointer; user-select: none;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
